### PR TITLE
set_nest_at_indices #1954

### PR DESCRIPTION
**Pull request by Bhavik-Ardeshna**
Task: Change the code of the set_nest_at_indices nest function. (Reformat nest https://github.com/unifyai/ivy/issues/793)

Changes:

- Added generic input argument to functions of Ivy
- Change the docstring of the function
- Run all tests for the same
- Added examples covering all cases

### CODE REVIEW

**Comment by Bhavik-Ardeshna** [comment]
@ankitdsi2010 Hi!

I have been facing this issue another time. I have also tested locally all the changes which I have done. The same issue happened in my previous PR.

Please address me to complete the PR.

Thank You

File: ivy/functional/ivy/nest.py [hunk 1]
```diff
@@ -168,7 +168,12 @@ def prune_nest_at_indices(nest, indices):
     [prune_nest_at_index(nest, index) for index in indices]
 
 
-def set_nest_at_indices(nest, indices, values):
+def set_nest_at_indices(
+    nest: Union[List, Tuple, ivy.Array, ivy.NativeArray],
```
Comment (ankitdsi2010) [inline ivy/functional/ivy/nest.py:172]: Add Dict type here

File: ivy/functional/ivy/nest.py [hunk 2]
```diff
@@ -168,7 +168,12 @@ def prune_nest_at_indices(nest, indices):
...
+    indices: Union[List[int], Tuple[int], Iterable[int]],
+    values: Union[List[int], Tuple[int], Iterable[int]],
+) -> Any:
+
```
Comment (ankitdsi2010) [inline ivy/functional/ivy/nest.py:176]: Remove the extra space here

File: ivy/functional/ivy/nest.py [hunk 3]
```diff
@@ -180,7 +185,46 @@ def set_nest_at_indices(nest, indices, values):
...
+    With :code:`Dict` input:
+
+    >>> x = {'a': [1., 2., 3.], 'b': [4., 5., 6.], 'c': [0.]}
+    >>> y = (('a', 1), ('b', 2), ('c', 0))
+    >>> values = (11., 22., 33.)
+    >>> ivy.set_nest_at_indices(nest, indices, values)
+    >>> print(nest)
+    {'a': [1.0, 11.0, 3.0], 'b': [4.0, 5.0, 22.0], 'c': [33.0]}
     """
+
```
Comment (ankitdsi2010) [inline ivy/functional/ivy/nest.py:227]: Remove the extra space here

**Review by ankitdsi2010 — CHANGES_REQUESTED** [review]
Go through the run-docstring-tests error logs and you will find under the captured stdout call, that for the dictionary case, the test fails, try to rectify that.

**Review by ankitdsi2010 — CHANGES_REQUESTED** [review]
then check if the tests are running, most probably they are failing because because the dict case output generated and expected are not matching which can be seen in the error logs like in https://github.com/unifyai/ivy/runs/7386327506?check_suite_focus=true#step:4:159

**Comment by Bhavik-Ardeshna** [comment]
> @ankitdsi2010 Resolved some issues but getting this `AssertionError` My DocString is changed as per your comments.
> 
> ```
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> =========================== short test summary info ============================
> FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[jax] - AssertionError: 
> FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[numpy] - AssertionEr...
> FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[tensorflow] - Assert...
> FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[torch] - AssertionEr...
> ======================= 4 failed, 41 warnings in 15.96s ========================
> Error: Process completed with exit code 1.
> ```

Hey! @ankitdsi2010 
I have to add submodule ```test_array_api``` to my clone repo to resolve the specific ```test-array-api-jax``` errors.

**Comment by ankitdsi2010** [comment]
Yeah I wasn't asking for the framework specific checks that's why, only for the test-docstrings check which is now running perfectly so it's fine.

**Comment by ankitdsi2010** [comment]
LGTM! Thanks

**Comment by Bhavik-Ardeshna** [comment]
@ankitdsi2010 
Thank you for your guidance and support. Looking forward to working with you.

**Pull request merged**